pages/fundamentals/constraints.mdx

```diff
@@ -161,6 +161,102 @@ DROP CONSTRAINT ON (n:Employee) ASSERT n.name, n.address IS UNIQUE;
 
 Now, `SHOW CONSTRAINT INFO;` returns an empty set.
 
+## Data type constraint
+
+The data type constraint enforces that each label-property pair is of a certain data type.
+
+<Callout type="info">
+
+Adding a data type constraint does not create a [label-property
+index](/fundamentals/indexes), it needs to be added manually.
+
+</Callout>
+
+The data type constraint can be enforced using the following language
+construct:
+
+```cypher
+CREATE CONSTRAINT ON (n:label) ASSERT n.property IS TYPED DATA_TYPE;
+```
+
+The supported data types are:
+
+| Data type           |
+| ------------------- |
+| `NULL`              |
+| `STRING`            |
+| `BOOLEAN`           |
+| `INTEGER`           |
+| `FLOAT`             |
+| `LIST`              |
+| `MAP`               |
+| `DURATION`          |
+| `DATE`              |
+| `LOCALTIME`         |
+| `LOCALDATETIME`     |
+| `ZONEDDATETIME`     |
+| `ENUM`              |
+| `POINT`             |
+
+To confirm that the constraint was successfully created use the following query:
+
+```cypher
+SHOW CONSTRAINT INFO;
+```
+
+Trying to modify the database in a way that violates the constraint will yield
+an error `IS TYPED DATA_TYPE violation on Label(property)`.
+
+Constraints are dropped using the `DROP` clause:
+
+```cypher
+DROP CONSTRAINT ON (n:label) ASSERT n.property IS TYPED DATA_TYPE;
+```
+
+You can only have one data type constraint on a given label-property pair.
+Attempting to create a second data type constraint on a given label-property pair will yield an error
+`Constraint IS TYPED DATA_TYPE on :Label(property) already exists`.
+Attempting to drop a data type constraint which doesn't exist will yield an error
+`Constraint IS TYPED DATA_TYPE on :Node(prop) doesn't exist`.
+
+<Callout type="info">
+
+Data type constraints are not yet supported in the [`schema.assert()`](/querying/schema#assert) procedure.
+
+</Callout>
+
+
+### Example
+
+If the database is used to hold basic information about a person like their name and surname
+you can enforce them to be strings by running the following queries:
+
+```cypher
+CREATE CONSTRAINT ON (n:Person) ASSERT n.name IS TYPED STRING;
+CREATE CONSTRAINT ON (n:Person) ASSERT n.surname IS TYPED STRING;
+```
+
+Then `SHOW CONSTRAINT INFO;` should return the following result:
+
+```
++-----------------+--------------+-----------------+-------------+
+| constraint type | label        | properties      |  data_type  |
++-----------------+--------------+-----------------+-------------|
+| data_type       | Person       | name            |   STRING    |
+| data_type       | Person       | surname         |   STRING    |
++-----------------+--------------+-----------------+-------------|
+```
+
+To drop the created constraints, use the following queries:
+
+```cypher
+DROP CONSTRAINT ON (n:Person) ASSERT n.name IS TYPED STRING;
+DROP CONSTRAINT ON (n:Person) ASSERT n.surname IS TYPED STRING;

```

Comment:
Don't repeat the STRING datatype in the 2nd example

Reply:
Changed with n.age and INTEGER